staging/src/k8s.io/apiserver/pkg/registry/generic/registry/storage_factory.go

```diff
@@ -82,25 +84,36 @@ func StorageWithCacher(capacity int) generic.StorageDecorator {
 // only from the test harness, so Register/Cleanup will be
 // no-op at runtime.
 
+var cleanupLock sync.Mutex
 var cleanup []func() = nil
 
 func TrackStorageCleanup() {
+	cleanupLock.Lock()
+	defer cleanupLock.Unlock()
+
+	if cleanup != nil {
+		panic("Conflicting storage tracking")
```

Comment:
Why not allow multiple cleanups and just only respect the first?

Reply:
We use the global state and have no knowledge to which "session" the storage belongs. The whole mechanism was a quickfix with this assumption in mind. Wiring through storage shutdown properly has been done by @frobware, but introduces quite some ugliness. So @dims added this hack for the tests. We still don't see complete stability, so this let's the mechanism fail early, but with a clear message.